### PR TITLE
UI Editor for `plant-status` card

### PR DESCRIPTION
**Pull request by iantrich**

### CODE REVIEW

File: src/panels/lovelace/cards/hui-plant-status-card.js [hunk 1]
```diff
@@ -2,7 +2,21 @@ import "../../../cards/ha-plant-card";
...
+};
+
 class HuiPlantStatusCard extends LegacyWrapperCard {
+  static async getConfigElement() {
+    await import("../editor/config-elements/hui-plant-status-card-editor");
+    return document.createElement("hui-plant-status-card-editor");
+  }
+
+  static getStubConfig() {
+    return {};
```
[the remark this answers is not in the document]
Comment (iantrich) [reply src/panels/lovelace/cards/hui-plant-status-card.js]: Put this in the editor as that already has `hass` wired up.

[the remark this answers is not in the document]
Comment (iantrich) [reply src/panels/lovelace/cards/hui-plant-status-card.js]: Oh...that won't work as the passed in stub is validated against the editor struct which now requires entity to be filled in...will need to do this in the stub after all

File: src/panels/lovelace/editor/config-elements/hui-plant-status-card-editor.ts [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+    this._config = { type: "plant-status", ...config };
+  }
+
+  static get properties(): PropertyDeclarations {
+    return { hass: {}, _config: {} };
+  }
+
+  get _entity(): string {
+    const entity = Object.keys(this.hass!.states).find(
```
Comment (balloob) [inline src/panels/lovelace/editor/config-elements/hui-plant-status-card-editor.ts]: We should do this in the stub.

Comment (iantrich) [reply src/panels/lovelace/editor/config-elements/hui-plant-status-card-editor.ts]: Yeah. I noted that I my comment. Just didn't have time last night to figure it out

Comment (iantrich) [reply src/panels/lovelace/editor/config-elements/hui-plant-status-card-editor.ts]: I can't get this to work in the stub. and always get an undefined hass object Tried in the light card as that is Typescript to check my work and can't get around this TS error:
```
property 'hass' does not exist on type 'typeof HuiLightCard'
```

File: src/panels/lovelace/cards/hui-plant-status-card.js [hunk 3]
```diff
@@ -2,7 +2,21 @@ import "../../../cards/ha-plant-card";
 
 import LegacyWrapperCard from "./hui-legacy-wrapper-card";
 
+export const Config = {
```
Comment (balloob) [inline src/panels/lovelace/cards/hui-plant-status-card.js]: Not how you define an interface.

Comment (iantrich) [reply src/panels/lovelace/cards/hui-plant-status-card.js]: Do you suggest I do it in a TS file then? This is what I found when googling for just JS interfaces as they don’t really exist in JS.

**Review by balloob — APPROVED** [review]

**Pull request merged**